DNN Platform/Library/Entities/Users/UserController.cs

```diff
@@ -543,7 +543,18 @@ public static UserInfo GetCachedUser(int portalId, string username)
             var masterPortalId = GetEffectivePortalId(portalId);
 
             var user = MembershipProvider.Instance().GetUserByUserName(masterPortalId, username);
-            FixMemberPortalId(user, portalId);
+            if (user != null && user.PortalID != portalId)

```

Comment:
I think this logic should go into the `FixMemberPortalId()` function, which is called multiple times in this class. Fixing this inline here, like this, allows the other usages of the function to stay broken (and the user object to have outdated roles.)